src/app/clusters/scenes-server/SceneHandlerImpl.cpp

```diff
@@ -16,10 +16,138 @@
+#include <app/util/endpoint-config-api.h>
 
 namespace chip {
 namespace scenes {
 
+namespace {
+
+using ConcreteAttributePath  = app::ConcreteAttributePath;
+using AttributeValuePairType = app::Clusters::ScenesManagement::Structs::AttributeValuePair::Type;
+
+/// IsAttributeTypeValid
+/// @brief Check if the attribute type is valid for a scene extension field value according to the spec.
+/// @param type Type of the attribute's metadata
+/// @return
+bool IsSceneValidAttributeType(EmberAfAttributeType type)
+{
+    return (type == ZCL_INT8U_ATTRIBUTE_TYPE || type == ZCL_INT16U_ATTRIBUTE_TYPE || type == ZCL_INT32U_ATTRIBUTE_TYPE ||
+            type == ZCL_INT64U_ATTRIBUTE_TYPE || type == ZCL_INT8S_ATTRIBUTE_TYPE || type == ZCL_INT16S_ATTRIBUTE_TYPE ||
+            type == ZCL_INT32S_ATTRIBUTE_TYPE || type == ZCL_INT64S_ATTRIBUTE_TYPE);
+}
+
+/// IsSignedAttribute
+/// @brief Compare the attribute type to the signed attribute types in ZCL and return true if it is signed.
+/// @param attributeType metadata's attribute type
+/// @return
+bool IsSignedAttribute(EmberAfAttributeType attributeType)
+{
+    return (attributeType >= ZCL_INT8S_ATTRIBUTE_TYPE && attributeType <= ZCL_INT64S_ATTRIBUTE_TYPE);
+}
+
+/// ConvertByteArrayToUInt64
+/// @brief Helper function to convert a byte array to a uint64_t value
+/// @param EmberAfDefaultAttributeValue & defaultValue
+/// @param len Length of the byte array
+/// @return uint64_t Value
+/// @note The attribute table supports 8 bytes only for unsigned integers, 4 bytes is the maximum for signed integers
+uint64_t ConvertByteArrayToUInt64(const EmberAfDefaultAttributeValue & defaultValue, uint16_t len)
+{
+    if (len <= 2)
+    {
+        return static_cast<uint64_t>(defaultValue.defaultValue);
+    }
+    else
+    {
+        uint64_t result     = 0;
+        const uint8_t * val = defaultValue.ptrToDefaultValue;
+        for (size_t i = 0; i < len; i++)
+        {
+            result = (result << 8) | val[(CHIP_CONFIG_BIG_ENDIAN_TARGET ? i : (len - 1) - i)];
+        }
+        return result;
+    }
+}
+
+/// CapAttributeID
+/// Cap the attribute value based on the attribute's min and max if they are defined,
+/// or based on the attribute's size if they are not.
+/// @param[in] aVPair   AttributeValuePairType
+/// @param[in] metadata  EmberAfAttributeMetadata
+///
+void CapAttributeID(AttributeValuePairType & aVPair, const EmberAfAttributeMetadata * metadata)
+{
+    // Calculate the maximum value that can be represented with the given number of bytes
+    uint64_t maxValue = 0;
+
+    // Check if the attribute type is signed
+    if (IsSignedAttribute(metadata->attributeType))
+    {
+        maxValue = (1ULL << ((emberAfAttributeSize(metadata) * 8) - 1)) - 1;
+    }
+    else
+    {
+        maxValue = (1ULL << (emberAfAttributeSize(metadata) * 8)) - 1;
+    }
+
+    // Check metadata for min and max values
+    if ((metadata->mask & ATTRIBUTE_MASK_MIN_MAX) && metadata->defaultValue.ptrToMinMaxValue)
```

Comment:
Please add an accessor to the metadata struct instead of raw mask checks.

Also, under what conditions would the mask flag be set but ptrToMinMaxValue be null?